src/unitxt/llm_as_judge.py

```diff
@@ -14,38 +14,119 @@
         reduction_map (dict): A dictionary specifying the reduction method for the metric.
         batch_size (int): The size of the bulk.
     """
 
     main_score: str = "llm_as_judge"
-    task: Literal[
-        "rating.single_turn",
-        "rating.single_turn_with_reference",
-        "pairwise_comparative_rating.single_turn",
-    ]
+    task: str
     template: Template
     system_prompt: SystemPrompt = Field(default_factory=EmptySystemPrompt)
     format: Format = Field(default_factory=SystemFormat)
-    strip_system_prompt_and_format_from_inputs: bool = True
     inference_model: InferenceEngine
     reduction_map: Optional[Dict[str, List[str]]] = None
     batch_size: int = 32
     prediction_type = Any  # Because handled with multiple tasks
 
+    def verify(self):
+        if not isinstance(self.template, Template):
+            raise ValueError(
+                f"Provided template argument to 'LLMAsJudge' metric is not of type Template, but {type(self.template)}"
+            )
+        if self.format and not isinstance(self.format, Format):
+            raise ValueError(
+                f"Provided format argument to 'LLMAsJudge' metric is not of type Format, but {type(self.format)}"
+            )
+
+        if self.system_prompt and not isinstance(self.system_prompt, SystemPrompt):
+            raise ValueError(
+                f"Provided system_prompt argument to 'LLMAsJudge' metric is not of type SystemPrompt, but {type(self.system_prompt)}"
+            )
+
+        if isinstance(self.inference_model, OpenAiInferenceEngine):
+            if self.format and type(self.format) is not SystemFormat:
+                raise ValueError(
+                    "Error in 'LLMAsJudge' metric. Inference model 'OpenAiInferenceEngine' does "
+                    "not support formatting. Please remove the format definition from the recipe"
+                    " (OpenAi Chat API take care of the formatting automatically)."
+                )
+            if self.system_prompt and type(self.system_prompt) is not EmptySystemPrompt:
+                raise ValueError(
+                    "Error in 'LLMAsJudge' metric. Inference model 'OpenAiInferenceEngine' does "
+                    "not support system prompt. Please remove the system_prompt definition from the recipe"
+                    " (Current implementation of Unitxt does not support this."
+                    " Support will be added in future updates)."
+                )
+
+    def get_full_task_name(self):
+        raise NotImplementedError
+        # f"tasks.response_assessment.{self.task}",
+
+    def compute(
+        self,
+        references: List[List[Any]],
+        predictions: List[Any],
+        task_data: List[Dict],
+    ) -> List[Dict[str, Any]]:
+        instances = self.prepare_instances(references, predictions, task_data)
+        outputs = self.infer_instances(instances)
+        return self.get_results_from_outputs(outputs)
```

Comment:
This method is not declared in this class as far as I can see. Add it with explanation of what it should do